docs/backend-system/building-plugins-and-modules/08-migrating.md

```diff
@@ -6,4 +6,223 @@ sidebar_label: Migration Guide
+        http.use(router);
+      },
+    });
+  },
+});
+```
+
+Done! Users of this plugin are now able to import the `kubernetesPlugin` and register it in their backend using
+
+```ts
+// packages/backend/src/index.ts
+import { kubernetesPlugin } from '@backstage/plugin-kubernetes-backend';
+backend.add(kubernetesPlugin);
+```
+
+There's one thing missing that those sharp eyed readers might have noticed, the `clusterSupplier` option is missing from the original plugin. Let's add it and discuss the alternatives.
+
+One alternative is pass the `ClusterSupplier` in as options to the plugin which is quick and easy but not very flexible and hard to evolve without introducing breaking changes as it changes the public API for the plugin. Having complex types passed in directly to the plugin also clutters the backend setup code and makes it harder to read.
+
+Options are primarily used for simple configuration values that are not complex types. In this case we want to allow users to register their own `ClusterSupplier` implementations to the plugin. This is where the new backend system's [ExtensionPoints](fixme.md) but let's look at doing this with options first.
+
+```ts
+/* omitted imports but they remain the same as above */
+
+export interface KubernetesOptions {
+  clusterSupplier?: KubernetesClustersSupplier;
+}
+
+const kubernetesPlugin = createBackendPlugin((options: KubernetesOptions) => ({
+  id: 'kubernetes',
+  register(env) {
+    env.registerInit({
+      deps: {
+        /* omitted imports but they remain the same as above */
+      },
+      async init({ config, logger, catalogApi, discovery, http }) {
+        const { router } = await KubernetesBuilder.createBuilder({
+          config,
+          logger,
+          catalogApi,
+          discovery,
+        })
+          .setClusterSupplier(options.clusterSupplier)
+          .build();
+        http.use(router);
+      },
+    });
+  },
+}));
+```
+
+The above would allow users to specify their own `ClusterSupplier` implementation to the plugin like this:
+
+```ts
+backend.add(
+  kubernetesPlugin({ clusterSupplier: new MyCustomClusterSupplier() }),
+);
+```
+
+Just to echo what was said above, this is not a very flexible solution and will for example be problematic to maintain backwards compatible if we start evolving the options to for example accept multiple suppliers or tweak the `ClusterSupplier` interface.
+
+The new [ExtensionPoints](fixme.md) API allows [Modules](fixme) to add function into the backend plugin itself, in this case an additional `ClusterSupplier`
+
+The kubernetes backend plugin only supports one `ClusterSupplier` at this time but let's look at how we could add support for multiple suppliers using extension points. This allows users to install several modules that add their own `ClusterSupplier` implementations to the plugin like this:
+
+```ts
+backend.add(kubernetesPlugin());
+backend.add(GoogleContainerEngineModule());
+backend.add(EKSModule());
```

Comment:
```suggestion
backend.add(googleContainerEngineModule());
backend.add(ekSModule());
```

also see ongoing discussion on how to best name these (especially in examples), let's keep it like this for now but maybe it could have the word Kubernetes in there just to clarify